src/secretlens/SecretLensFunction.ts

```diff
@@ -14,37 +14,75 @@ export class SecretLensFunction implements interfaces.ISecretLensFunction {
         this.provider = provider;
     }
 
-    public encrypt(inputText: string): string {
-        var encrypted: string = "";
-        var saltedPassword = this.password;
-        if (this.useSalt) {
-            var salt = crypto.randomBytes(this.saltSize).toString('hex');
-            saltedPassword = salt + this.password;
-        }
-        const cipher = crypto.createCipher('aes-256-cbc', saltedPassword);
-        encrypted = cipher.update(inputText, 'utf8', 'hex');
-        encrypted += cipher.final('hex');
-        if (this.useSalt) {
-            return salt + encrypted;
+    public encrypt(inputText: string, cryptoMethod: string): string {
+
+        if (cryptoMethod == "pbkdf2") {
+
+            // Equivalent shell command: openssl enc -aes-256-cbc -pbkdf2 | xxd -p | tr -d "\n"
+            //
+            var binsalt = crypto.randomBytes(8);
+            let derivedKey = crypto.pbkdf2Sync(this.password, binsalt, 10000, 48, 'sha256');
+            let key = derivedKey.slice(0, 32);
+            let iv = derivedKey.slice(32, 48);
+
+            let cipher = crypto.createCipheriv('aes-256-cbc', key, iv)
+            var encrypted = cipher.update(inputText, 'utf8', 'hex');
+            encrypted += cipher.final('hex');
+            return 'pbkdf2:53616c7465645f5f' + binsalt.toString('hex') + encrypted;
```

Comment:
What is the purpose of the string "53616c7465645f5f" after pbkdf2?

Reply:
Hi @fcrespo82!

53616c7465645f5f is the hex encoding for string "Salted__".
This is added for compatibility with the openssl command.
Specifically, openssl uses a salt generation algorithm where the first 8 bytes are always "Salted__" and the last five bytes are random.

I wanted compatibility with openssl so that in case secretlens gets broken, this gives an easy way to still be able to decrypt secrets independently.

Reply:
Ok, thanks for the explanation.